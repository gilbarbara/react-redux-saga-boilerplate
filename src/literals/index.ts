import { keyMirror } from '@gilbarbara/helpers';

import { Status } from '~/types';

export const ActionTypes = keyMirror({
  ALERTS_HIDE: undefined,
  ALERTS_SHOW: undefined,
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
});

export const STATUS: Status = {
  IDLE: 'idle',
  RUNNING: 'running',
  READY: 'ready',
  SUCCESS: 'success',
  ERROR: 'error',
};
