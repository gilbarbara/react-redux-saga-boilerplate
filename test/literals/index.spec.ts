import { STATUS } from '~/literals';

describe('literals', () => {
  it('should match the snapshot', () => {
    expect(STATUS).toMatchSnapshot();
  });
});
