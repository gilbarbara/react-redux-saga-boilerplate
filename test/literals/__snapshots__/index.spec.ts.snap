// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`literals > should match the snapshot 1`] = `
{
  "ALERTS_HIDE": "ALERTS_HIDE",
  "ALERTS_SHOW": "ALERTS_SHOW",
  "GITHUB_GET_REPOS_FAILURE": "GITHUB_GET_REPOS_FAILURE",
  "GITHUB_GET_REPOS_REQUEST": "GITHUB_GET_REPOS_REQUEST",
  "GITHUB_GET_REPOS_SUCCESS": "GITHUB_GET_REPOS_SUCCESS",
  "SET_APP_OPTIONS": "SET_APP_OPTIONS",
  "USER_LOGIN_FAILURE": "USER_LOGIN_FAILURE",
  "USER_LOGIN_REQUEST": "USER_LOGIN_REQUEST",
  "USER_LOGIN_SUCCESS": "USER_LOGIN_SUCCESS",
  "USER_LOGOUT_FAILURE": "USER_LOGOUT_FAILURE",
  "USER_LOGOUT_REQUEST": "USER_LOGOUT_REQUEST",
  "USER_LOGOUT_SUCCESS": "USER_LOGOUT_SUCCESS",
}
`;

exports[`literals > should match the snapshot 2`] = `
{
  "ERROR": "error",
  "IDLE": "idle",
  "READY": "ready",
  "RUNNING": "running",
  "SUCCESS": "success",
}
`;
